Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM osrf/ros:humble-desktop-full
```

Comment:
@ryan-roche meta: consider using a multi-stage Dockerfile to only carry over what's strictly necessary to run the driver (and leave behind all development tooling like `colcon`).

Reply:
Could you explain a bit more what exactly that would entail? I'm not too familiar with multi-stage Dockerfiles, but they look incredibly promising.

Reply:
My idea for this Dockerfile was that it would be used as a base for projects that use the Spot driver, hence using the `desktop-full` image for ROS2 Humble. The image would include the full ROS Humble install and the precompiled driver, so all they need to do to get going with their project is mount their code into the container and build _their_ packages.

I would argue that there wouldn't be much of a reason to use the driver just on its own when things like the raw Python SDK and Tablet Controller exist- nearly everyone using the ROS driver is using it because they're working on a ROS project of their own that adds additional nodes that work with the driver, and if everyone using this image would have to manually extend it by installing the dev tools that we intentionally omitted, we might as well just leave them in the original image.

I suppose we could also make separate tags of the image that include different amounts of additional tools? One that includes the full ROS Humble install with tools like `colcon`, and another that, as you said, only contains what is strictly necessary to run the driver?

Reply:
> My idea for this Dockerfile was that it would be used as a base for projects that use the Spot driver, hence using the desktop-full image for ROS2 Humble. The image would include the full ROS Humble install and the precompiled driver, so all they need to do to get going with their project is mount their code into the container and build their packages.

I see. That sounds very different from what this PR description suggests:

> ...creating a dockerfile that can be used to run the driver in **production**...

I'll re-review considering that what you are actually proposing is distributing `spot_ros2` in a Docker image for downstream users to build on top.

Reply:
Nevermind my first comment. A multi-stage build is overkill for the current setup.

Reply:
I see the merits in your interpretation, and I apologize if my proposal was unclear. I had assumed that it was acceptable as-is given my previous discussions. I had used the term production primarily to emphasize that this is different from the devcontainer dockerfile.

It's your repo, so you get the final say. I just wanted to explain my line of thinking. Separate tags is probably the best plan of action- that way we can have both options.

Reply:
I've updated the PR description to better reflect my line of thinking. I believe that there are merits to both approaches, and that it shouldn't be too difficult to build both described images and just push them with different tags (i.e. `minimal` vs `project-base`)